securedrop_client/gui/widgets.py

```diff
@@ -1806,61 +1915,129 @@ def __init__(
         self.error.hide()
 
         self.bubble_area_layout.addWidget(self.error)
+        self.sender_icon.show()
 
         message_succeeded_signal.connect(self._on_reply_success)
         message_failed_signal.connect(self._on_reply_failure)
+        self.controller.update_authenticated_user.connect(self._on_update_authenticated_user)
+        self.controller.authentication_state.connect(self._on_authentication_changed)
 
-        if reply_status == "SUCCEEDED":
-            self.set_normal_styles()
-            self.error.hide()
-        elif reply_status == "FAILED":
-            self.set_failed_styles()
-            self.error.show()
-        elif reply_status == "PENDING":
-            self.set_pending_styles()
+        self.sender_icon.is_current_user = self._sender_is_current_user
+        if self._sender:
+            self.sender_icon.initials = self._sender.initials
+
+        self._update_styles()
+
+    @property
+    def sender_is_current_user(self) -> bool:
+        return self._sender_is_current_user
+
+    @sender_is_current_user.setter
+    def sender_is_current_user(self, sender_is_current_user: bool) -> None:
+        if self._sender_is_current_user != sender_is_current_user:
+            self._sender_is_current_user = sender_is_current_user
+            self.sender_icon.is_current_user = sender_is_current_user
+
+    @property
+    def sender(self) -> User:
+        return self._sender
+
+    @sender.setter
+    def sender(self, sender: User) -> None:
+        if self._sender != sender:
+            self._sender = sender
+
+        if self._sender:
+            self.sender_icon.initials = self._sender.initials
+
+    @pyqtSlot(bool)
+    def _on_authentication_changed(self, authenticated: bool) -> None:
+        """
+        When the user logs out, update the reply badge.
+        """
+        if not authenticated:
+            self.sender_is_current_user = False
+            self._update_styles()
+
+    @pyqtSlot(User)
+    def _on_update_authenticated_user(self, user: User) -> None:
+        """
+        When the user logs in or updates user info, update the reply badge.
+        """
+        if user.uuid == self.sender.uuid:
+            self.sender_is_current_user = True
+            self.sender = user
+            self._update_styles()
 
     @pyqtSlot(str, str, str)
-    def _on_reply_success(self, source_id: str, message_uuid: str, content: str) -> None:
+    def _on_reply_success(self, source_uuid: str, uuid: str, content: str) -> None:
         """
         Conditionally update this ReplyWidget's state if and only if the message_uuid of the emitted
         signal matches the uuid of this widget.
         """
-        if message_uuid == self.uuid:
-            self.set_normal_styles()
-            self.error.hide()
+        if self.uuid == uuid:
+            self.status = "SUCCEEDED"  # TODO: Add and use success status in db.ReplySendStatusCodes
```

Comment:
Yes, `db.ReplySendStatusCodes` only includes 'FAILED' and 'PENDING' statuses. I see other areas of the code that have to hardcode an implied 'SUCCEEDED' status, so I followed suit and made note of how we can improve this: we can add a success status to `db.ReplySendStatusCodes` and start tracking successful replies with a non-null status. This will require several unrelated updates so I left it out of this PR.